google/cloud/sql/connector/instance.py

```diff
@@ -303,6 +302,11 @@ async def _refresh_task(self: RefreshAheadCache, delay: int) -> ConnectionInfo:
                     await asyncio.sleep(delay)
                 refresh_task = asyncio.create_task(self._perform_refresh())
                 refresh_data = await refresh_task
+                # check that refresh is valid
+                if not await _is_valid(refresh_task):
+                    raise RefreshNotValidError(
+                        f"['{self._instance_connection_string}']: Invalid refresh operation. Certficate appears to be expired."
+                    )
```

Comment:
we do this in AlloyDB: https://github.com/GoogleCloudPlatform/alloydb-python-connector/blob/bb5d107b801e713b4e94c8dc644e6950717704cb/google/cloud/alloydb/connector/instance.py#L196-L199

This check fixed a test that was passing due to bad legacy test infra.